src/SearchComponent/SearchComponent.tsx

```diff
@@ -0,0 +1,39 @@
+import { useState } from 'react';
+import styles from './SearchComponent.module.css';
+
+interface onSearchProps {
+  onSearch: (term: string) => void;
+}
+
+const SearchComponent = ({ onSearch }: onSearchProps) => {
+  const [localSearchTerm, setLocalSearchTerm] = useState(
+    localStorage.getItem('starTrek_searchTerm') || ''
```

Comment:
better to make this component less dependent on localStorage so it can be reused